src/Compilers/Core/Portable/Text/SubText.cs

```diff
@@ -22,9 +22,9 @@ public SubText(SourceText text, TextSpan span)
                 throw new ArgumentNullException(nameof(text));
             }
 
+            // span.Start and span.End are valid wrt eachother by nature of being passed in as a TextSpan,
+            // so there is no need to verify span.Start against text.Length or span.End against zero.
```

Comment:
I don't follow how this is true. I can absolutely construct a bad `TextSpan` and pass it in, what prevents me from doing that?

Reply:
a textspan itself prevents the end from being before the start.  so a text span's start/end always represent at least a legal range where end >= start.   given that.  you only ened to validate that the start of this text span is not before the start of the text, and the end of the text span is not after the end of the text.

Reply:
Thanks for the clarification!